katharsis-client/src/main/java/io/katharsis/client/internal/RelationshipRepositoryStubImpl.java

```diff
@@ -0,0 +1,126 @@
+		extends AbstractStub implements RelationshipRepositoryStub<T, T_ID, D, D_ID> {
+
+	private Class<T> sourceClass;
+	private Class<D> targetClass;
+	private ResourceInformation resourceInformation;
+	private RegistryEntry<?> relationshipEntry;
+
+	public RelationshipRepositoryStubImpl(KatharsisClient client, Class<T> sourceClass, Class<D> targetClass,
+			ResourceInformation resourceInformation, RequestUrlBuilder urlBuilder, RegistryEntry<?> relationshipEntry) {
+		super(client, urlBuilder);
+		this.sourceClass = sourceClass;
+		this.targetClass = targetClass;
+		this.resourceInformation = resourceInformation;
+		this.relationshipEntry = relationshipEntry;
+	}
+
+	@Override
+	public void setRelation(T source, D_ID targetId, String fieldName) {
+		Serializable sourceId = getSourceId(source);
+		HttpUrl url = urlBuilder.buildUrl(sourceClass, sourceId, null, fieldName);
+		execute(url, "PATCH", targetId);
+	}
+
+	@Override
+	public void setRelations(T source, Iterable<D_ID> targetIds, String fieldName) {
+		Serializable sourceId = getSourceId(source);
+		HttpUrl url = urlBuilder.buildUrl(sourceClass, sourceId, null, fieldName);
+		execute(url, "PATCH", targetIds);
+	}
+
+	@Override
+	public void addRelations(T source, Iterable<D_ID> targetIds, String fieldName) {
+		Serializable sourceId = getSourceId(source);
+		HttpUrl url = urlBuilder.buildUrl(sourceClass, sourceId, null, fieldName);
+		execute(url, "POST", targetIds);
+	}
+
+	@Override
+	public void removeRelations(T source, Iterable<D_ID> targetIds, String fieldName) {
+		Serializable sourceId = getSourceId(source);
+		HttpUrl url = urlBuilder.buildUrl(sourceClass, sourceId, null, fieldName);
+		execute(url, "DELETE", targetIds);
+	}
+
+	private Serializable getSourceId(T source) {
+		ResourceField idField = resourceInformation.getIdField();
+		return (Serializable) PropertyUtils.getProperty(source, idField.getUnderlyingName());
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public D findOneTarget(T_ID sourceId, String fieldName, QueryParams queryParams) {
+		HttpUrl url = urlBuilder.buildUrl(sourceClass, sourceId, queryParams, fieldName);
+		BaseResponseContext responseContext = executeGet(url);
+		return (D) responseContext.getResponse().getEntity();
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public List<D> findManyTargets(T_ID sourceId, String fieldName, QueryParams queryParams) {
+		HttpUrl url = urlBuilder.buildUrl(sourceClass, sourceId, queryParams, fieldName);
+		BaseResponseContext responseContext = executeGet(url);
+		return (List<D>) responseContext.getResponse().getEntity();
+	}
+
+	private void execute(HttpUrl requestUrl, String method, Object targetIds) {
+		JsonPath fieldPath = new ResourcePath(resourceInformation.getResourceType());
+
+		JsonApiResponse response = new JsonApiResponse();
+		BaseResponseContext context;
+		if (targetIds instanceof Iterable) {
+			ArrayList<LinkageContainer> containers = new ArrayList<LinkageContainer>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)